File: WEEK04/1965 상자넣기/원영.py
# 가장 긴 증가하는 수열(LIS)과 같은 알고리즘이다.
import sys
input = sys.stdin.readline

n = int(input())
nums = list(map(int ,input().split()))

answer = [1]*n

for i in range(n):
  for j in range(i):
    if nums[i] > nums[j]:
      answer[i] = max(answer[i], answer[j]+1)

print(max(answer))

File: WEEK05/238 자신을 제외한 배열의 곱/원영.js
/**
 * @param {number[]} nums 
 * @return {number[]}
 */
var productExceptSelf = function (nums) {
  let p = 1;
  answer = [];
  nums.map((el) => {
    answer.push(p);
    p *= el;
  });
  p = 1;
  for (let i = answer.length - 1; i > -1; i--) {
    answer[i] = p * answer[i];
    p *= nums[i];
  }
  return(answer);
};